pkg/chain/ethereum/ethutil/signer_test.go

```diff
@@ -0,0 +1,165 @@
+package ethutil
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+func TestSignAndVerify(t *testing.T) {
+	signing, err := newSigning()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	message := []byte("He that breaks a thing to find out what it is, has " +
+		"left the path of wisdom.")
+
+	signature, err := signing.Sign(message)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var tests = map[string]struct {
+		message                 []byte
+		signature               []byte
+		validSignatureExpected  bool
+		validationErrorExpected bool
+	}{
+		"valid signature for message": {
+			message:                 message,
+			signature:               signature,
+			validSignatureExpected:  true,
+			validationErrorExpected: false,
+		},
+		"invalid signature for message": {
+			message:                 []byte("I am sorry"),
+			signature:               signature,
+			validSignatureExpected:  false,
+			validationErrorExpected: false,
+		},
+		"corrupted signature": {
+			message:                 message,
+			signature:               []byte("I am so sorry"),
+			validSignatureExpected:  false,
+			validationErrorExpected: true,
+		},
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			ok, err := signing.Verify(test.message, test.signature)
+
+			if !ok && test.validSignatureExpected {
+				t.Errorf("expected valid signature but verification failed")
+			}
+			if ok && !test.validSignatureExpected {
+				t.Errorf("expected invalid signature but verification succeeded")
+			}
+
+			if err == nil && test.validationErrorExpected {
+				t.Errorf("expected signature validation error; none happened")
+			}
+			if err != nil && !test.validationErrorExpected {
+				t.Errorf("unexpected signature validation error [%v]", err)
+			}
+		})
+	}
+}
+
+func TestSignAndVerifyWithProvidedPublicKey(t *testing.T) {
+	message := []byte("I am looking for someone to share in an adventure")
+
+	signing1, err := newSigning()
```

Comment:
Looks like `signing` was used before you created a struct `EthereumSigner`. It's up to you, but we can name it ex. `signer1` and `signer2` below.

Reply:
Done in 7acbaeada69dd654a500baae55f1601b2a524bed